procedures/crates.md

```diff
@@ -0,0 +1,33 @@
+# Crate Policies
+This document describe the guidelines for splitting the compiler up into crates and using
+third-party crates in the compiler. These guidelines were originally discussed at the Rust All
+Hands 2019 by the compiler team and others.
+
+## Where should compiler crates live?
+As library-ification progresses and parts of the compiler are extracted into separate reusable
+crates, these new crates should be hosted in the `rust-lang` organization.
+
+## Who owns these crates?
+It is desired that a compiler team member has loose ownership over a crate so that there is someone
+who is responsible for making sure that new versions are published and that pull requests are
+reviewed.
+
+## What should these crates be named?
+When naming a crate, it is worth considering that a `rustc_` prefix could result in
+compiler-specific changes being made where this isn't appropriate as the crate was actually
+intended to be general-purpose.
+
+## What should be included in a out-of-tree crate?
+A maintenance policy should be included to set appropriate expectations for anyone who may wish to
```

Comment:
Is there a link to an example that could be added here?

Reply:
I've added a small example.